### DISCUSSION TITLE
Table support

### DISCUSSION THREAD

**Discussion by javiavid** [aidenlx/media-extended#221] [Ideas]
Do you think it is interesting that when creating the captures they are ordered in a table?
Visually I think it is better, but I don't know if it would be practical.

![image](https://github.com/PKM-er/media-extended/assets/7627504/bfea6620-95bc-4265-bce6-e78481819584)

**Comment by javiavid** [comment]
If subtitles are integrated in the future, perhaps it could also go in the table. You could also include the metadata.


<img width="542" alt="image" src="https://github.com/PKM-er/media-extended/assets/7627504/166f48b1-80ae-4b18-b053-1b2cb8db55b2">

**Comment by d0verenny** [comment]
Visually it looks great, however, the markdown table aren't very comfy to work with. I'm thinking the similar could be achieved by Dataviewjs, but I don't know how, yet :) Maybe with tags or something. 

I personally used ranges with opening and closing tags ones for the (not quite) similar task in the plugin called "Projects". I needed a template that would grab my review from certain notes and render them in the table in the Projects' UI. More details here in [this post](https://github.com/Acylation/obsidian-projects/issues/2#issue-3614246666)

Below I provide not a "ready-made" solution, but an example of how you can use open and closing "fake tags" (i call them "delimiters" below). Maybe that could help the dev, or yourself. You could use dataviewjs (Dataview plugin), and may (or may not) achieve what you want, since it renders the tables, I don't know. 

You would still need to write the text normally, and the table would render only in read/preview mode after the caret is gone. And note that it wouldn't "hide" the text by itself. It can render that table in a separate dataviewjs query. So that's a quirk with I see now with achieving it via Dataview.

So, er, yeh, probably the native plugin's ability to do that would be nice, even though I personally don't consider it to be of the highest priority, tbh.

A quote from the post with the script I "vibecoded" and used:

> ```
> ---
> type: 
> subType: ""
> title: 
> englishTitle: 
> year: ""
> dataSource: 
> url: 
> id: 
> developers: 
> publishers: 
> genres: 
> onlineRating: 
> image: "![]({{ image }})"
> released: 
> releaseDate: 
> played: 
> personalRating: 
> dataviewCode: |
>   ```dataviewjs
>   // === Wait for the frontmatter to populate ===
>   await new Promise(resolve => setTimeout(resolve, 500));
> 
>   let noteName = tp.frontmatter.englishTitle;
>   let notePath = `Media DB/games/${noteName}.md`;
> 
>   let heading = "## Review";
>   let delimiter = "***";
> 
>   let file = app.vault.getAbstractFileByPath(notePath);
> 
>   if (!file) {
>       dv.paragraph("_File not found:_ " + notePath);
>   } else {
>       (async () => {
>           let content = await app.vault.read(file);
>           let lines = content.split(/\r?\n/);
> 
>           let inSection = false;
>           let sectionLines = [];
> 
>           for (let line of lines) {
>               if (!inSection) {
>                   if (line.trim() === heading) inSection = true;
>               } else {
>                   if (line.trim() === delimiter) break;
>                   sectionLines.push(line);
>               }
>           }
> 
>           if (sectionLines.length) {
>               dv.paragraph(sectionLines.join("\n").trim());
>           } else {
>               dv.paragraph("_No " + heading + " section found in note:_ " + notePath);
>           }
>       })();
>   }
> tags:
> ---
> 
> <%* 
> await new Promise(resolve => setTimeout(resolve, 1000)); // wait 0.5s
> 
> let headingTitle = tp.frontmatter.englishTitle; // takes the name from frontmatter englishTitle
> tR += `# ${headingTitle}`;
> %>
> 
> ***
> 
> videoUrl::![](<%*
> await new Promise(resolve => setTimeout(resolve, 1000)); // wait 0.5s
> 
> let ytTitle = tp.frontmatter.englishTitle; // YouTube title. takes the name from frontmatter englishTitle
> let query = encodeURIComponent(ytTitle);
> 
> let apiKey = "ENTER_YOUR_YOUTUBE_API_KEY!";
> let url = `https://www.googleapis.com/youtube/v3/search?part=snippet&type=video&q=${query}&maxResults=1&key=${apiKey}`;
> 
> let response = await fetch(url);
> let data = await response.json();
> 
> let vidId = data.items?.[0]?.id?.videoId ?? "";
> tR += `https://www.youtube.com/watch?v=${vidId}`;
> %>)
> 
> ***
> 
> ## Review
> 
> Check check
> 
> ***
> 
> ```dataviewjs
> // === Wait for the note title to be ready (0.5s) ===
> await new Promise(resolve => setTimeout(resolve, 1000));
> 
> let noteName = tp.frontmatter.englishTitle; 
> let notePath = `Media DB/games/${noteName}.md`;
> 
> let heading = "## Review"; 
> let delimiter = "***";
> 
> let file = app.vault.getAbstractFileByPath(notePath);
> 
> if (!file) {
>   dv.paragraph("_File not found:_ " + notePath);
> } else {
>   (async () => {
> 	  let content = await app.vault.read(file);
> 	  let lines = content.split(/\r?\n/);
> 
> 	  let inSection = false;
> 	  let sectionLines = [];
> 
> 	  for (let line of lines) {
> 		  if (!inSection) {
> 			  if (line.trim() === heading) inSection = true;
> 		  } else {
> 			  if (line.trim() === delimiter) break;
> 			  sectionLines.push(line);
> 		  }
> 	  }
> 
> 	  if (sectionLines.length) {
> 		  dv.paragraph(sectionLines.join("\n").trim());
> 	  } else {
> 		  dv.paragraph("_No " + heading + " section found in note:_ " + notePath);
> 	  }
>   })();
> }
> ```

#### What the script does exactly?

##### 1. Searching for youtube video based on the englishTitle in the front matter and pastes it in the embeded format `![](youtube.com/yourvideo`). 

If you want to change it to something else, for example, the file name, change this variable `let ytTitle = tp.frontmatter.englishTitle;` it to `let ytTitle = tp.frontmatter.title` or `tp.file.title` (not sure if the latter one will work, but you can try.

##### 2. Makes a dataviewjs query from the body of the document, between delimiters `## Review` and `***` (straight line).

**Why not `---`?**

To avoid any conflicts. It's cleaner that way. You can change delimiters to your custom ones in the code. Just change them in these lines after `=`:

`let heading = "## Review"; `
`let delimiter = "***";`

> ⚠️ heading and delimeter are open and closing tags here I was talking about at the beginning
